lib/document/stateTransition/validation/data/validateDocumentsUniquenessByIndicesFactory.js

```diff
@@ -33,19 +34,40 @@ function validateDocumentsUniquenessByIndicesFactory(stateRepository) {
         documentSchema.indices
           .filter((index) => index.unique)
           .forEach((indexDefinition) => {
-            const where = indexDefinition.properties
-              .map((property) => {
-                const propertyName = Object.keys(property)[0];
+            const where = [];
 
-                let propertyValue;
-                if (propertyName === '$ownerId') {
+            indexDefinition.properties.forEach((property) => {
+              const propertyName = Object.keys(property)[0];
+              let propertyValue;
+
+              switch (propertyName) {
+                case '$ownerId':
                   propertyValue = ownerId;
-                } else {
-                  propertyValue = documentTransition.getData()[propertyName];
+                  break;
+                case '$createdAt':
+                  if (documentTransition.getAction()
+                    === AbstractDocumentTransition.ACTIONS.CREATE) {
+                    const createdAt = documentTransition.getCreatedAt();
```

Comment:
Do we want all the documents in the batch to have the same timestamp?

Reply:
Ah I forgot that we now have documenTransition and batch transition is a different thing